### PR TITLE
Added `queryParams` to the `HttpRequestInformation` annotation.

### PR DESCRIPTION
**Pull request by vcolin7**
Fixes: #40377.

Also renamed `requestHeaders` to `headers`.

### CODE REVIEW

File: sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/http/rest/SwaggerMethodParser.java [hunk 1]
```diff
@@ -148,6 +149,38 @@ public SwaggerMethodParser(Method swaggerMethod) {
...
+            for (final String queryParam : requestQueryParams) {
+                final int equalsIndex = queryParam.indexOf("=");
+
+                if (equalsIndex >= 0) {
+                    final String paramName = queryParam.substring(0, equalsIndex).trim();
+
+                    if (!paramName.isEmpty()) {
+                        String paramValue = queryParam.substring(equalsIndex + 1).trim();
+
+                        if (!paramValue.isEmpty()) {
```
Comment (alzimmermsft) [inline sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/http/rest/SwaggerMethodParser.java:164]: Query parameter values should be allowed to be empty and having leading and trailing spaces.

File: sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/http/rest/SwaggerMethodParser.java [hunk 2]
```diff
@@ -148,6 +149,38 @@ public SwaggerMethodParser(Method swaggerMethod) {
...
+                            constantQueryParams.compute(paramName, (key, value) -> {
+                                if (value == null) {
```
Comment (alzimmermsft) [inline sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/http/rest/SwaggerMethodParser.java:166]: Can simplify this to have this null check if just initialize the value list, that way we only need to write the list add logic once.

File: sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/http/rest/SwaggerMethodParser.java [hunk 3]
```diff
@@ -148,6 +149,38 @@ public SwaggerMethodParser(Method swaggerMethod) {
...
+                                    List<String> valueList = new ArrayList<>();
+
+                                    valueList.add(paramValue.replace(",", "%2C"));
```
Comment (alzimmermsft) [inline sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/http/rest/SwaggerMethodParser.java:169]: We'll want to use proper query parameter value encoding here, should replace this with `UrlEscapers.QUERY_ESCAPER.escape`

File: sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/http/rest/SwaggerMethodParser.java [hunk 4]
```diff
@@ -72,6 +72,7 @@ public class SwaggerMethodParser implements HttpResponseDecodeData {
     private final ClientLogger methodLogger;
     private final HttpMethod httpMethod;
     private final String relativePath;
+    private final Map<String, List<String>> constantQueryParams = new HashMap<>();
```
Comment (alzimmermsft) [inline sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/http/rest/SwaggerMethodParser.java:75]: Use a `LinkedHashMap` to preserve order, it'll be safer in the long run

File: sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/http/rest/SwaggerMethodParser.java [hunk 5]
```diff
@@ -327,10 +362,19 @@ public String setPath(Object[] methodArguments, ObjectSerializer serializer) {
     @SuppressWarnings("unchecked")
     public void setEncodedQueryParameters(Object[] swaggerMethodArguments, UrlBuilder urlBuilder,
                                           ObjectSerializer serializer) {
+        // First we add the constant query parameters.
+        for (Map.Entry<String, List<String>> queryParam : constantQueryParams.entrySet()) {
+            for (String value : queryParam.getValue()) {
+                addSerializedQueryParameter(serializer, value, false, urlBuilder, queryParam.getKey());
```
Comment (alzimmermsft) [inline sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/http/rest/SwaggerMethodParser.java:368]: This should use `urlBuilder.addQueryParameter` directly as

1. We shouldn't need to serialize the header value as the annotation only allows string values, so there is no object value to serialize.
2. We should already have the query parameter value encoded when we parse them as constants.

File: sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/http/rest/SwaggerMethodParser.java [hunk 6]
```diff
@@ -148,6 +149,38 @@ public SwaggerMethodParser(Method swaggerMethod) {
             }
         }
 
+        final String[] requestQueryParams = httpRequestInformation.queryParams();
+
+        if (requestQueryParams != null) {
+            for (final String queryParam : requestQueryParams) {
+                final int equalsIndex = queryParam.indexOf("=");
+
+                if (equalsIndex >= 0) {
```
Comment (srnagar) [inline sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/http/rest/SwaggerMethodParser.java:158]: Where do we do error handling when the query params are not in the right format? We should not fail silently when the annotations are invalid.

**Review by alzimmermsft — APPROVED** [review]

**Review by srnagar — APPROVED** [review]

**Pull request merged**